### PR TITLE
Add dashboard thumbnails

### PR DESCRIPTION
**Pull request by Melissa0x1f992**
Add dashboard thumbnails using the /y22/projects/:uuid/bitmap endpoint.
Make the /projects page much more responsive

### CODE REVIEW

File: pages/projects.js [hunk 1]
```diff
@@ -3,37 +3,49 @@ import Router from "next/router";
...
-	static Joining = new MembershipStatus("Joining...");
-	static Joined = new MembershipStatus("Joined");
-	static Leaving = new MembershipStatus("Leaving...");
-	static NotJoined = new MembershipStatus("Not Joined");
-	static Unknown = new MembershipStatus("Unknown");
+	static Joining = new MembershipStatus("Joining...", "🔃");
+	static Joined = new MembershipStatus("Joined", "✅");
+	static Leaving = new MembershipStatus("Leaving...", "🔃");
+	static NotJoined = new MembershipStatus("Not Joined", "❌");
+	static Unknown = new MembershipStatus("Unknown", "❔");
```
Comment (haykam821) [inline pages/projects.js:38]: Could Font Awesome icons be used instead of emojis here?

File: pages/projects.js [hunk 2]
```diff
@@ -80,27 +92,41 @@ const ProjectPreview = ({ project }) => {
...
+				{typeof membershipStatus !== MembershipStatus.Unknown && (
+					<form onSubmit={membershipStatus === MembershipStatus.Joined ? leaveProject : joinProject}>
+						<SubmitButton
+							type="submit"
+							value={
+								membershipStatus === MembershipStatus.Joined ? "➖ Leave" :
+								membershipStatus === MembershipStatus.Joining ? "➕ Joining..." :
+								membershipStatus === MembershipStatus.NotJoined ? "➕ Join" :
+								membershipStatus === MembershipStatus.Leaving ? "➖ Leaving..." :
+								"Error"
```
Comment (haykam821) [inline pages/projects.js:124]: These values could be stored in the `MembershipStatus` instances.

**Comment by Megadynamite** [comment]
This pull request is for an event that has ended and therefore I will close it. Should it become relevant again, it should be reopened.

**Comment by haykam821** [comment]
I think the projects page has diverged too much for this pull request to remain relevant. A new design should be built with #31 and #32 in mind.

**Pull request closed without merging**